plansys2_bt_actions/include/plansys2_bt_actions/BTActionNode.hpp

```diff
@@ -134,32 +142,74 @@ class BtActionNode : public BT::ActionNodeBase
   {
     // first step to be done only at the beginning of the Action
     if (status() == BT::NodeStatus::IDLE) {
-      createActionClient(action_name_);
+      // Get the required items from the blackboard
+      unsigned int server_timeout_int;
+      if (!getInput<unsigned int>("server_timeout", server_timeout_int)) {
+        // This will only happen if `providedPorts` is overridden and
+        // the child class does not provide the "server_timeout" port.
+        // Child classes can use the `providedBasicPorts` method to avoid
+        // this issue
+        RCLCPP_INFO(
+          node_->get_logger(),
+          "Missing input port [server_timeout], "
+          "using default value of 1s");
+        RCLCPP_DEBUG(
+          node_->get_logger(),
+          "Use the `providedBasicPorts` method to avoid this issue");
+        server_timeout_int = 1000;
+      }
+      server_timeout_ = std::chrono::milliseconds(server_timeout_int);
+
+      if (!createActionClient(action_name_)) {
+        RCLCPP_ERROR(node_->get_logger(), "Could not create action client");
+        return BT::NodeStatus::FAILURE;
+      }
 
       // setting the status to RUNNING to notify the BT Loggers (if any)
       setStatus(BT::NodeStatus::RUNNING);
 
       // user defined callback
       on_tick();
+      if (return_failure_) {
```

Comment:
Why to check here? `on_tick` could change it? In this case, it could also return bool